src/Providers/Http/Exception/ResponseException.php

```diff
@@ -4,13 +4,85 @@
 
 namespace WordPress\AiClient\Providers\Http\Exception;
 
-use Exception;
+use Psr\Http\Client\ClientExceptionInterface;
+use WordPress\AiClient\Common\Exception\RuntimeException;
+use WordPress\AiClient\Providers\Http\DTO\Response;
 
 /**
  * Exception class for HTTP response errors.
  *
+ * This is used when response data is unexpected or malformed,
+ * typically indicating that a provider changed in ways our code
+ * is not aware of or when parsing response data fails.
+ *
  * @since 0.1.0
  */
-class ResponseException extends Exception
+class ResponseException extends RuntimeException implements ClientExceptionInterface
 {
+    /**
+     * Creates a ResponseException for missing expected data.
+     *
+     * @since n.e.x.t
+     *
+     * @param string $apiName The name of the API/provider.
+     * @param string $fieldName The field that was expected but missing.
+     * @param string $context Additional context about where the field was expected.
+     * @return self
+     */
+    public static function fromMissingData(string $apiName, string $fieldName, string $context = ''): self
```

Comment:
There are several other places where this should be used:
* All exceptions thrown in `AbstractOpenAiCompatibleTextGenerationModel::parseResponseToGenerativeAiResult`
* All exceptions thrown in `AbstractOpenAiCompatibleTextGenerationModel::parseResponseChoiceToCandidate`
* All exceptions thrown in `AbstractOpenAiCompatibleTextGenerationModel::parseResponseChoiceMessageParts`
* All exceptions thrown in `AbstractOpenAiCompatibleImageGenerationModel::parseResponseToGenerativeAiResult`
* All exceptions thrown in `AbstractOpenAiCompatibleImageGenerationModel::parseResponseChoiceToCandidate`

Reply:
For some of these where it's about something more specific than a _missing_ field, we could have another `public static function fromInvalidData(string $apiName, string $message): self` here.

Reply:
Looks like this has been resolved. 👍